src/Generator/Generators/CSharp/CSharpTypePrinter.cs

```diff
@@ -655,6 +655,9 @@ public override TypePrinterResult VisitParameter(Parameter param, bool hasName)
             typeBuilder.Append(printedType.NameSuffix);
             var type = typeBuilder.ToString();
 
+            if (param.QualifiedType.Type.GetFinalQualifiedPointee().Type.Desugar() is FunctionType)
```

Comment:
For simplicity just check if the dictionary contains the param. If it does, there's no way it got there in the first place unless it has a function type.